lib/plugins/aws/provider.js

```diff
@@ -1036,6 +1036,11 @@ class AwsProvider {
                       },
                     },
                   },
+                  dependencies: {
+                    command: ['entryPoint'],
+                    entryPoint: ['command'],
+                    workingDirectory: ['entryPoint', 'command'],
+                  },
```

Comment:
Can you confirm what warning what is shown to user if that is not met (?) -> So far we avoided to use this constructs in schema (as result error reporting is vague), and handled this manually in implementation.

Reply:
Sure, I was checking that out before and below you can find example outputs:

```
Serverless: Configuration warning at 'functions.greeter.image': should have property entryPoint when property command is present
```

```
Serverless: Configuration warning:
Serverless:   at 'functions.greeter.image': should have properties entryPoint, command when property workingDirectory is present
Serverless:   at 'functions.greeter.image': should have properties entryPoint, command when property workingDirectory is present
Serverless:
```

I agree that in case of two (or more) properties missing, the message isn't perfect (it reports the same warning twice, the same one for each missing dependency).

I guess handling it directly in implementation and throwing an appropriate error is a valid approach as well :+1:. Let me know what do you think

Reply:
I think warnings looks good, but I would definitely filter so they're not doubled. I think we can improve our errors filter by adding `filterDuplicateErrors` here: https://github.com/serverless/serverless/blob/fd9b26a9e898685da81663064274250c5771363c/lib/classes/ConfigSchemaHandler/normalizeAjvErrors.js#L255

Reply:
great idea - added an extra commit which filters out duplicate errors based on message